Dockerfile

```diff
@@ -1,46 +1,60 @@
-    rm -rf utilitywarehouse
-
-# Clone golangci repo reset to the latest tag and install
-RUN mkdir golangci && cd golangci && git clone https://github.com/golangci/golangci-lint.git && \
-    cd golangci-lint && git reset --hard $(git rev-list --tags --max-count=1) && \
-    go install github.com/golangci/golangci-lint/cmd/golangci-lint && \
-    cd ../../ && rm -rf golangci
+FROM golang:1.12.4
+
+ENV PROTOBUF_VERSION="3.7.0" \
+    GOLANG_PROTOBUF_VERSION="1.3.1" \
+    GOGO_PROTOBUF_VERSION="1.2.1" \
+    VALIDATE_PROTOBUF_VERSION="0.0.14" \
+    VALIDATORS_PROTOBUF_VERSION="1f388280e944c97cc59c75d8c84a704097d1f1d6" \
+    UWPARTNER_PROTOBUF_VERSION="de4552500027969912fd801dcc5269a153b3fffe" \
+    GOLANGCI_LINT_VERSION="1.16.0" \
+    GOTESTSUM_VERSION="0.3.4"
+
+## Dependencies
+RUN apt-get update && \
+    apt-get install -y ca-certificates curl unzip tar
+
+## `protoc` binary
+RUN mkdir -p /tmp/protoc && cd /tmp/protoc && \
+    curl -sSL https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip -o protoc.zip && \
+    unzip protoc.zip && \
+    mv /tmp/protoc/bin/protoc* /usr/local/bin && \
+    mv /tmp/protoc/include /usr/local/include && \
+    rm -rf /tmp/protoc
+
+## `protoc-gen-go` binaries
+## `protoc-gen-gogoslick` binaries
+RUN GO111MODULE=on go get \
+    github.com/golang/protobuf/protoc-gen-go@v${GOLANG_PROTOBUF_VERSION} \
+    github.com/gogo/protobuf/protoc-gen-gogoslick@v${GOGO_PROTOBUF_VERSION} && \
+    mv /go/bin/protoc-gen-* /usr/local/bin/
+
+## `protoc-gen-validate` binary
+## `protoc-gen-govalidators` binary
+RUN GO111MODULE=on go get \
+    github.com/envoyproxy/protoc-gen-validate@v${VALIDATE_PROTOBUF_VERSION} \
+    github.com/mwitkow/go-proto-validators/protoc-gen-govalidators@${VALIDATORS_PROTOBUF_VERSION} && \
+    mv /go/bin/protoc-gen-* /usr/local/bin/
+
+## `protoc-gen-uwpartner` binary
+RUN GO111MODULE=on go get \
+    github.com/utilitywarehouse/protoc-gen-uwpartner@${UWPARTNER_PROTOBUF_VERSION} && \
+    mv /go/bin/protoc-gen-* /usr/local/bin/
+
+## `golangci-lint` binary
+# Golangci Lint
+RUN mkdir -p /tmp/golangci-lint && cd /tmp/golangci-lint && \
+    curl -sSL https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz -o golangci-lint.tar.gz && \
+    tar -xvf golangci-lint.tar.gz && \
+    mv /tmp/golangci-lint/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64/golangci-lint /usr/local/bin && \
+    rm -rf /tmp/golangci-lint
+ADD ./.golangci.yml /
+
+## `gotestsum` binary
+RUN curl -sSL https://github.com/gotestyourself/gotestsum/releases/download/v${GOTESTSUM_VERSION}/gotestsum_${GOTESTSUM_VERSION}_linux_amd64.tar.gz | \
+    tar -xz -C /usr/local/bin gotestsum
 
 # Copy in makefile and project docker image
 WORKDIR /build
 ADD ./Makefile .
 ADD ./Dockerfile.project .
-ADD ./.golangci.yml .
```

Comment:
Were you planning something different with this? The Makefile still uses it

Reply:
Check further up, it's still included.

Golangci-lint searches multiple paths, I've just moved it out of the working project directory.

`docker run -it partner-docker-go:latest golangci-lint -v`

Reply:
https://github.com/utilitywarehouse/partner-docker-go-circleci-build/pull/7/files#diff-3254677a7917c6c01f55212f86c57fbfR50

Reply:
Ah sorry!

Reply:
Wait, the old path is directly referenced in the Makefile?

Reply:
I fixed that in further commits